src/components/DonationForm/DonationForm.jsx

```diff
@@ -1,54 +1,145 @@
-import FurnitureField from '../FurnitureField/FurnitureField';
+// import FurnitureField from '../FurnitureField/FurnitureField';
+import DropZone from '../DropZone/DropZone';
+import { sendEmail } from '../../utils/utils';
+import dconfirmemailtemplate from '../EmailTemplates/dconfirmemailtemplate';
+import ImageDetails from '../ImageDetails/ImageDetails';
+
+const itemFieldSchema = {
+  itemName: yup.string().required('A Furniture Selection is Required'),
+};
 
 const schema = yup.object({
   firstName: yup.string().required('Invalid first name'),
-  lastName: yup.string().required('Invalid first name'),
-  zipcode: yup.number().typeError('ZIP code must be a number'),
+  lastName: yup.string().required('Invalid last name'),
+  phoneNumber: yup
+    .string()
+    .required('Phone number is required')
+    .matches(/[0-9]{3}-[0-9]{3}-[0-9]{4}/, 'Phone number must match xxx-xxx-xxxx format'),
+  zipcode: yup
+    .string()
+    .required('ZIP Code is required')
+    .matches(/^[0-9]+$/, 'ZIP Code must be a number')
+    .min(5, 'ZIP Code must be exactly 5 digits')
+    .max(5, 'ZIP Code must be exactly 5 digits'),
   email1: yup.string().email('Invalid email').required('Email required'),
-  email2: yup.string().email('Invalid email').required('Email required'),
+  email2: yup
+    .string()
+    .required('Email required')
+    .oneOf([yup.ref('email1'), null], 'Emails must both match'),
+  Items: yup.array().of(yup.object().shape(itemFieldSchema), 'A Furniture Selection is Required'),
 });
 
 function DonationForm() {
   const {
+    handleSubmit,
     register,
     control,
-    handleSubmit,
     formState: { errors },
   } = useForm({
     resolver: yupResolver(schema),
   });
 
+  const [furnitureOptions, setFurnitureOptions] = useState([
+    'Dressers',
+    'Clean Housewares',
+    'Antiques',
+    'Art',
+    'Clean rugs',
+    'Home Decor items',
+    'Pet care items',
+    'Patio Furniture',
+  ]);
+
   const {
-    fields: furnitureFields,
-    append: appendFurniture,
-    remove: removeFurniture,
+    fields: DonatedFurnitureList,
+    append: appendDonatedFurniture,
+    remove: removeDonatedFurniture,
   } = useFieldArray({
     control,
-    name: 'furnitureField',
+    name: 'DonatedFurniture',
   });
 
+  // const {
+  //   fields: imageDetailList,
+  //   append: appendImage,
+  //   remove: removeImage,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'ImageDetail',
+  // });
+
+  // const {
+  //   fields: itemsList,
+  //   append: appendItem,
+  //   remove: removeItem,
+  // } = useFieldArray({
+  //   control,
+  //   name: 'Items',
+  // });
+
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const [files, setFiles] = useState([]);
+  const [filesIntermediate, setFilesIntermediate] = useState([]);
+  // const [images, setImages] = useState([]);
+
+  const onSubmit = async data => {
+    const urls = await Promise.all(files.map(async file => uploadImage(file)));
```

Comment:
🚫 **[eslint]** <[no-undef](https://eslint.org/docs/rules/no-undef)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'uploadImage' is not defined.